force_wfmanager/ui/setup/mco/kpi_specification_view.py

```diff
@@ -0,0 +1,308 @@
+                   'kpi_names'
+    )
+
+    # ------------------
+    #      Buttons
+    # ------------------
+
+    #: Adds selected_non_kpi variable as a new MCO KPI
+    add_kpi_button = Button('New KPI')
+
+    #: Removes selected_kpi from the MCO
+    remove_kpi_button = Button('Delete KPI')
+
+    # ------------------
+    #       View
+    # ------------------
+
+    def default_traits_view(self):
+
+        # TableEditor to display non_kpi_variables
+        table_editor = TableEditor(
+            columns=[
+                ObjectColumn(name="name",
+                             label="name",
+                             resize_mode="stretch"),
+                ObjectColumn(name="type",
+                             label="type",
+                             resize_mode="stretch")
+            ],
+            auto_size=False,
+            selected='selected_non_kpi'
+        )
+
+        # ListEditor to display kpi_model_views
+        list_editor = ListEditor(
+            page_name='.label',
+            use_notebook=True,
+            dock_style='tab',
+            selected='selected_kpi',
+            style='custom'
+        )
+
+        traits_view = View(
+            VGroup(
+                VGroup(
+                    Item('kpi_model_views',
+                         editor=list_editor,
+                         show_label=False,
+                         style='custom',
+                         ),
+                    show_labels=False,
+                    show_border=True
+                ),
+                HGroup(
+                    UReadonly(
+                        'non_kpi_variables',
+                        editor=table_editor
+                    ),
+                    show_border=True,
+                    label='Non-KPI Variables'
+                ),
+                HGroup(
+                    Item('add_kpi_button',
+                         springy=True),
+                    Item('remove_kpi_button'),
+                    show_labels=False,
+                ),
+            ),
+            dock='fixed',
+            kind="livemodal"
+        )
+
+        return traits_view
+
+    #: Property getters
+    @cached_property
+    def _get_kpi_names(self):
+        """Listens to model.kpis to extract model names for display"""
+        kpi_names = []
+
+        for kpi in self.model.kpis:
+            kpi_names.append(kpi.name)
+
+        return kpi_names
+
+    @cached_property
+    def _get_non_kpi_variables(self):
+        """Listens to variable_names_registry to extract names
+         able to be assigned to kpis"""
+        non_kpi = []
+
+        if self.variable_names_registry is not None:
+            variables_stack = (self.variable_names_registry
+                               .available_variables_stack)
+            for execution_layer in variables_stack:
+                for variable in execution_layer:
+                    kpi_check = variable[0] not in self.kpi_names
```

Comment:
Any new KPI must be unique (not listed in the existing `kpi_names`) and only refer to an output variable in the process to optimise (exist in `variable_names_registry.data_source_outputs`). By enforcing this requirement upon creation, it seems more appropriate to remove the ability of the user to change the KPI name after instantiation.